doc/source/tutorials/instrument.rst

```diff
@@ -1,35 +1,34 @@
 How to add a new instrument in Qibolab?
 =======================================
 
-Currently, Qibolab support various instruments:
-as **controller**:
+Currently, Qibolab supports various **controller** instruments:
 
 * Quantum Machines
 * Zurich instruments
 
-and as **local oscillators**:
+and the following **local oscillators**:
 
 * Rhode&Schwartz
 * Erasynth++
 
 If you need to add a new driver, to support a new instruments in your setup, we encourage you to have a look at ``qibolab.instruments`` for complete examples.
 In this section, anyway, a basic example will be given.
 
-For clarity, we divide the instruments in two different groups: the **controllers** and the standard **instruments**, where the controller is an instrument that can execute pulse sequences.
+For clarity, we divide the instruments in two different groups: the **controllers** and the standard **instruments**,
+where the controller is an instrument that can execute pulse sequences.
 For example, a local oscillator is just an instrument, while Quantum Machines is a controller.
 
 Add an instrument
 -----------------
 
-The base of an instrument is :class:`qibolab.instruments.abstract.Instrument`.
+The base of an instrument is :class:`qibolab.instruments.abstract.Instrument`,
+which is a pydantic ``Model``.
 To accomodate different kind of instruments, a flexible interface is implemented
-with four abstract methods that are required to be implemented in the child
+with three abstract methods that are required to be implemented in the child
```

Comment:
If I remember correctly, while working on this, I opened another branch on top to remove `Instrument.setup()`. However, I eventually gave up mainly due to it being used in https://github.com/qiboteam/qibolab/blob/ff05ddbd8a0a30ccdc2c2e36b05b73489156d2dc/src/qibolab/platform/platform.py#L279-L281

which is usually needed for setting up external LO parameters (eg. for TWPA pumps) and in other similar cases. 

At this stage, I believe `setup()` is only needed for "auxilliary" instruments, not `Controller`s, but I'd be more confident saying it after seeing Qblox in the new structure. Because I believe, `setup` was never actually used by any instrument other than Qblox and LO-like ones. There may even be a way to get rid of it completely, but for the interest of time and other open issues I didn't put much effort.

Reply:
Ok, I see the point: while with controller we have a communication through `play()`, the other instruments would be otherwise configured only upon creation, and this would be a bit unpractical (e.g. to change the frequency of an LO, after platform instantiation).

In this sense, it could even be useful in the long term, but I'd suggest not making it abstract: let's keep the default implementation empty, such that it will be only implemented by those instruments for which it has a non-trivial action.

Reply:
> Ok, I see the point: while with controller we have a communication through `play()`, the other instruments would be otherwise configured only upon creation, and this would be a bit unpractical (e.g. to change the frequency of an LO, after platform instantiation).

Indeed, the main reason we cannot do `setup` during creation is that it requires communication to the physical instrument in order to "upload" the settings, however during creation we are not yet connected. I believe we decided to do it this way to postpone connection as much as possible so that we are able to cover more code in the CI.

> In this sense, it could even be useful in the long term, but I'd suggest not making it abstract: let's keep the default implementation empty, such that it will be only implemented by those instruments for which it has a non-trivial action.

`setup` was converted from abstract to an empty method in 7209fef4548eab1fdcbea6a85738a860765ae465. Currently it is only implemented by the `LocalOscillator` instrument.